driver-core/src/test/unit/com/mongodb/internal/connection/InternalStreamConnectionInitializerSpecification.groovy

```diff
@@ -243,6 +256,94 @@ class InternalStreamConnectionInitializerSpecification extends Specification {
                                  [true, false]].combinations()
     }
 
+    @IgnoreIf({ !serverVersionAtLeast(4, 3) })
+    def 'should speculatively authenticate with default authenticator'() {
```

Comment:
Can you create async versions of these tests?  Perhaps using a Spock `where` clause?